### DISCUSSION TITLE
Merging Audio and Video Tracks with Captions

### DISCUSSION THREAD

**Discussion by andysingal** [openai/whisper#1691] [Q&A]
I was not able to add transcripts within the video, output mp4 is in audio format
```
Here is the code for reference:
https://colab.research.google.com/drive/1pRFlp1-OVZmRwyGLlMpexe6tSKVAKmwG?usp=sharing 
```
he issue is, i am able to get the audio from youtube video when merging i would like transcript within the video similar to Karoke. I was following https://www.digitalocean.com/community/tutorials/how-to-make-karaoke-videos-using-whisper-and-spleeter-ai-tools.
Please let me know if this assists , I do not understand if this is an issue with whisper or something else

**Comment by phineas-pta** [comment]
how can u expect to have video output when u downloaded only the audio without video 😅

**Reply by andysingal** [reply]
Thanks for your reply, I used a YouTube video.. yt-whisper to extract captions . I shared the code, how would you do it differently. Thanks

**Reply by phineas-pta** [reply]
download the whole video not just the audio

**Reply by andysingal** [reply]
Thank you, so download .mp3 and .webm?

**Reply by andysingal** [reply]
tried another youtube video, but the same issue. Can you please check: https://colab.research.google.com/drive/1pRFlp1-OVZmRwyGLlMpexe6tSKVAKmwG?usp=sharing?

**Reply by phineas-pta** [reply]
u still download audio only without video

**Comment by andysingal** [comment]
I tried pytube and other shared which only downloads .mp3 , dunno why this
happening

**Reply by phineas-pta** [reply]
change it to download webm or mp4 or mkv
